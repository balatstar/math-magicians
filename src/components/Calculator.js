import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';
import Display from './Display';

function Calculator() {
  const [calculatorState, setCalculatorState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonName) => {
    const newState = calculate(calculatorState, buttonName);
    setCalculatorState(newState);
  };

  return (
    <div className="calculator">
      <Display value={calculatorState.next || calculatorState.total || '0'} />
      <button type="button" className="key function" onClick={() => handleButtonClick('AC')}>AC</button>
      <button type="button" className="key function" onClick={() => handleButtonClick('+/-')}>+/-</button>
      <button type="button" className="key function" onClick={() => handleButtonClick('%')}>%</button>
      <button type="button" className="key operator" onClick={() => handleButtonClick('/')}>/</button>
      <button type="button" className="key digit" onClick={() => handleButtonClick('7')}>7</button>
      <button type="button" className="key digit" onClick={() => handleButtonClick('8')}>8</button>
      <button type="button" className="key digit" onClick={() => handleButtonClick('9')}>9</button>
      <button type="button" className="key operator" onClick={() => handleButtonClick('x')}>x</button>
      <button type="button" className="key digit" onClick={() => handleButtonClick('4')}>4</button>
      <button type="button" className="key digit" onClick={() => handleButtonClick('5')}>5</button>
      <button type="button" className="key digit" onClick={() => handleButtonClick('6')}>6</button>
      <button type="button" className="key operator" onClick={() => handleButtonClick('-')}>-</button>
      <button type="button" className="key digit" onClick={() => handleButtonClick('1')}>1</button>
      <button type="button" className="key digit" onClick={() => handleButtonClick('2')}>2</button>
      <button type="button" className="key digit" onClick={() => handleButtonClick('3')}>3</button>
      <button type="button" className="key operator" onClick={() => handleButtonClick('+')}>+</button>
      <button type="button" className="key digit zero" onClick={() => handleButtonClick('0')}>0</button>
      <button type="button" className="key digit" onClick={() => handleButtonClick('.')}>.</button>
      <button type="button" className="key operator" onClick={() => handleButtonClick('=')}>=</button>
    </div>
  );
}

export default Calculator;
