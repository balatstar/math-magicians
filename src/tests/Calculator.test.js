import * as React from 'react';
import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  it('renders Calculator component', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});