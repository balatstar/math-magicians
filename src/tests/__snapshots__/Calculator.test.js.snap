// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders Calculator component 1`] = `
<div>
  <h2>
    Let's do some math!
  </h2>
  <div
    class="calculator"
  >
    <div
      class="display"
    >
      0
    </div>
    <button
      class="key function"
      type="button"
    >
      AC
    </button>
    <button
      class="key function"
      type="button"
    >
      +/-
    </button>
    <button
      class="key function"
      type="button"
    >
      %
    </button>
    <button
      class="key operator"
      type="button"
    >
      รท
    </button>
    <button
      class="key digit"
      type="button"
    >
      7
    </button>
    <button
      class="key digit"
      type="button"
    >
      8
    </button>
    <button
      class="key digit"
      type="button"
    >
      9
    </button>
    <button
      class="key operator"
      type="button"
    >
      x
    </button>
    <button
      class="key digit"
      type="button"
    >
      4
    </button>
    <button
      class="key digit"
      type="button"
    >
      5
    </button>
    <button
      class="key digit"
      type="button"
    >
      6
    </button>
    <button
      class="key operator"
      type="button"
    >
      -
    </button>
    <button
      class="key digit"
      type="button"
    >
      1
    </button>
    <button
      class="key digit"
      type="button"
    >
      2
    </button>
    <button
      class="key digit"
      type="button"
    >
      3
    </button>
    <button
      class="key operator"
      type="button"
    >
      +
    </button>
    <button
      class="key digit zero"
      type="button"
    >
      0
    </button>
    <button
      class="key digit"
      type="button"
    >
      .
    </button>
    <button
      class="key operator"
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`QuoteDisplay renders QuoteDisplay component 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
